sunpy/map/mapbase.py

```diff
@@ -1482,7 +1483,14 @@ def rotate(self, angle: u.deg = None, rmatrix=None, order=4, scale=1.0,
         pad_x = int(np.max((diff[1], 0)))
         pad_y = int(np.max((diff[0], 0)))
 
-        new_data = np.pad(self.data,
+        if issubclass(self.data.dtype.type, numbers.Integral):
+            warnings.warn("Integer input data has been cast to float64.",
```

Comment:
Someone might correct me, but I don't think we need to add a new warning here, as the data is always cast to a `float` when `rotate` is used.

Reply:
An explicit warning is given by ```sunpy.image.transform``` when converted to float, considering this step is skipped, I thought of adding the warning here.

Reply:
Ah, that makes sense.